pkg/splunk/client/awss3client.go

```diff
@@ -125,10 +123,15 @@ func InitAWSClientSession(region, accessKeyID, secretAccessKey string) SplunkAWS
 }
 
 // NewAWSS3Client returns an AWS S3 client
-func NewAWSS3Client(bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, endpoint string, fn GetInitFunc) (S3Client, error) {
+func NewAWSS3Client(bucketName string, accessKeyID string, secretAccessKey string, prefix string, startAfter string, region string, endpoint string, fn GetInitFunc) (S3Client, error) {
 	var s3SplunkClient SplunkAWSS3Client
 	var err error
-	region := GetRegion(endpoint)
+
+	// for backward compatibility, if `region` is not specified in the CR,
+	// then derive the region from the endpoint.
+	if region == "" {
+		region = GetRegion(endpoint)
```

Comment:
What happens if region is not specified AND region can't be extracted from the endpoint .. should we return error? In other words, is region mandatory?

Reply:
Good point, let me make the change to return error in case we are not able to extract the region.